src/libs/actions/Report.js

```diff
@@ -323,6 +333,81 @@ function fetchChatReportsByIDs(chatList) {
         });
 }
 
+/**
+ * Given IOU and chat report ID fetches most recent IOU data from API and update stored data in Onyx.
+ *
+ * @param {Number} iouReportID
+ * @param {Number} chatReportID
+ */
+function updateIOUReportData(iouReportID, chatReportID) {
+    API.Get({
+        returnValueList: 'reportStuff',
+        reportIDList: iouReportID,
+        shouldLoadOptionalKeys: true,
+        includePinnedReports: true,
+    }).then((response) => {
+        if (!response) {
+            return;
+        }
+
+        if (response.jsonCode !== 200) {
+            console.error(response.message);
+            return;
+        }
+
+        const iouReportData = response.reports[iouReportID];
+        if (!iouReportData) {
+            console.error(`No iouReportData found for reportID ${iouReportID}`);
+            return;
+        }
+        const iouReportObject = getSimplifiedIOUReport(iouReportData, chatReportID);
+        const chatReportObject = {
+            hasOutstandingIOU: iouReportObject.stateNum === 1
+                && iouReportObject.total !== 0,
+            iouReportID,
+        };
+        if (!chatReportObject.hasOutstandingIOU) {
+            chatReportObject.iouReportID = null;
+        }
+        const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportObject.reportID}`;
+        const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`;
+        Onyx.merge(iouReportKey, iouReportObject);
+        Onyx.merge(reportKey, chatReportObject);
+    });
+}
+
+/**
+ * Given chat report ID finds active IOU report ID via GetIOUReport API call
+ *
+ * @param {Number} chatReportID
+ * @returns {Promise}
+ */
+function fetchIOUReportID(chatReportID) {
+    const chatReport = lodashGet(allReports, chatReportID);
+    const participants = chatReport.participants;
```

Comment:
Noticed one more bug here. In the case where an IOUAction is sent between two users who have not previously sent chat comments, then chatReport is undefined and causes an error here. 

- Create/find two new users who have not communicated with each other
- Create an IOU Request from A to B
- User B receives the push, but [chatReport is null here](https://github.com/Expensify/Expensify.cash/blob/975c0fc4bd158de51cf9e3c8e3fee2d91c5a81af/src/libs/actions/Report.js#L387)
- Unable to continue to get IOUReport, error logged

Thankfully it should be possible to get the name of the sender via `reportActionsToMerge.lastActorEmail` data [persisted to Onyx here](https://github.com/Expensify/Expensify.cash/blob/main/src/libs/actions/Report.js#L417). 

![Screenshot 2021-04-19 at 15 26 22](https://user-images.githubusercontent.com/10736861/115253503-590f4680-a124-11eb-94b0-52e65e6c2332.png)

Reply:
Actually, as `fetchIOUReportID` is only called if the report is not available locally, I think we can modify the function to always take `reportAction.actorEmail` as a parameter (and update the function name).

Pleas let me know if you have any further questions!

Reply:
I updated function to get debtor email as a parameter.

Reply:
This method works for user B. But when user A receives an update, we need to pass user B's email address. But report action doesn't have user B's email address here. What do you recommend when current user is the creator of fresh IOU report?

Reply:
Okay, it seems like adding the IOUReportID to the push notification data would save us a lot of trouble here. Let me take a look and get back to you. Perhaps I can resolve this in the follow up issue.

Reply:
Okay, so no need to worry about this issue. 

I'll use `pushJSON.originalMessage.transactionID` to retrieve the IOUReportID, using a [function that is already defined here](https://github.com/Expensify/Expensify.cash/blob/ccf9b7f0b158b4ae7d7ef3831f861018139aab90/src/libs/actions/IOU.js#L112-L131).